### PR TITLE
[fix] Synchronize closing the callback channel to avoid panics

### PR DESCRIPTION
**Pull request by cedric-cordenier**
Move each timer to its own goroutine, and add a `transmitCh` to synchronize all transmissions and associated channel closures. This avoids races with respect to the various channels used in the capability.

In particular, the use of `store.evict()` in the transmit handler guarantees that we'll only ever issue one transmission response and prevents races between the reporting plugin transmission and a request's expiration.

Also do some cleanup of the store: previously we set the request expiry there, but since none of the other logic around expiry happens there, I've moved this to the capability.

### CODE REVIEW

**Review by jmank88 — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 1]
```diff
@@ -199,14 +205,30 @@ func (o *capability) loop() {
...
+			o.handleTransmitMsg(ctx, resp)
 		}
 	}
 }
 
-func (o *capability) expiryWorker(ctx context.Context, r *request) {
+func (o *capability) handleTransmitMsg(ctx context.Context, resp *response) {
+	req, wasPresent := o.store.evict(ctx, resp.WorkflowExecutionID)
+	if !wasPresent {
+		o.lggr.Debugf("failed to clean up request with id %s", resp.WorkflowExecutionID)
```
Comment (bolekk) [inline pkg/capabilities/consensus/ocr3/capability.go:220]: We will see this log for every single request, right? IIUC we never cancel expiry timers early.

File: pkg/capabilities/consensus/ocr3/models.go [hunk 2]
```diff
@@ -13,14 +13,14 @@ type request struct {
 	ExpiresAt    time.Time
 
 	CallbackCh chan<- capabilities.CapabilityResponse
+	CleanupCh  chan struct{}
```
Comment (bolekk) [inline pkg/capabilities/consensus/ocr3/models.go:16]: I can't see this used anywhere...?

**Review by bolekk — COMMENTED** [review]
Looks good!

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 3]
```diff
@@ -199,14 +205,31 @@ func (o *capability) loop() {
...
+		case resp := <-o.transmitCh:
+			o.handleTransmitMsg(ctx, resp)
 		}
 	}
 }
 
-func (o *capability) expiryWorker(ctx context.Context, r *request) {
+func (o *capability) handleTransmitMsg(ctx context.Context, resp *response) {
+	req, wasPresent := o.store.evict(ctx, resp.WorkflowExecutionID)
+	if !wasPresent {
```
Comment (bolekk) [inline pkg/capabilities/consensus/ocr3/capability.go:219]: The debug log that was here can still be useful, I just thought that the message should be different if this is something expected. Or maybe no log for now is better.

**Review by bolekk — APPROVED** [review]

**Pull request merged**